src/mesh/mesh_adaptation.cpp

```diff
@@ -3,22 +3,21 @@
 namespace PHiLiP {
 
 template <int dim, typename real, typename MeshType>
-MeshAdaptation<dim,real,MeshType>::MeshAdaptation(double critical_res_input, int total_ref_cycle, double refine_frac, double coarsen_frac)
-    : critical_residual(critical_res_input)
-    , total_refinement_cycles(total_ref_cycle)
+MeshAdaptation<dim,real,MeshType>::MeshAdaptation(std::shared_ptr< DGBase<dim, real, MeshType> > dg)
```

Comment:
Like my comment in the ODE solver, this dg only seems to be used to pass parameters and when initializing the factory class. The other functions in MeshAdaptation also require dg to be passed in as arguments. Is there anything that prevents a pointer to dg from being stored within MeshAdaptation instead?

Reply:
Same as above.